### ISSUE TITLE
Uncaught (in promise) TypeError: track.artists is undefined

### ISSUE THREAD

**Issue by bluepuma77** [mopidy/mopidy.js#37]
For hours I have been trying to get mopidy.js to run, my major challenges:

1. Does not build with node 15 (seems to work with 12 and 14)
2. Parcel needs to be installed
3. Changing the script to connect to remote mopidy
4. Configure mopidy to enable remote hosts to access
5. Finally "Uncaught (in promise) TypeError: track.artists is undefined"

It seems to work now, I play a radio stream on the Raspi3+ (latest Raspberry Pi OS Lite), can access remotely via Iris and mopidy.js seems to connect, I can see web socket messages. But the console shows the exception and it displays "Unknown - Unknown".

![MopidyJS-track artists_is_undefined-2](https://user-images.githubusercontent.com/5221187/103158126-d3edd380-47ba-11eb-93c1-4d9bb0a3fffe.png)

**Comment by bluepuma77** [comment]
I need to add that 3) is well documented: "Make sure the Mopidy config http/allowed_origins includes pi3.local:1234"

I just had the challenge that a basic `sudo apt install mopidy` install would not read `/etc/mopidy/mopidy.conf`. It would tell me it reads the file, no error message, but then would not for example enable Iris.

Using a "professional" script does the trick:
```
# install mopidy with Pimoroni script
curl https://raw.githubusercontent.com/pimoroni/pirate-audio/master/mopidy/install.sh | sudo bash
# roll back some changes :)
sudo sed -i '/hifiberry/d' /boot/config.txt
sudo sed -i '/\[pidi\]/,+3d' /etc/mopidy/mopidy.conf
```


.

**Comment by bluepuma77** [comment]
MP3 stream does not work, MP3 local file does not work, but Spotify does work.

**Comment by kingosticks** [comment]
Is there an actual problem/question regarding Mopidy.js here? If not, I'm going to close this. 

If you are having problems with playing mp3 files or Mopidy in general then we can work on that at discuss.mopidy.com (you can also search for the similar issues on there already, it's likely a missing gstreamer plugin package).

**Comment by bluepuma77** [comment]
Yes, Mopidy.js throws an exception when playing MP3 streams or local MP3 files. 

`Uncaught (in promise) TypeError: track.artists is undefined`

This prevents the UI from updating the displayed information. It's a bug.

**Comment by kingosticks** [comment]
`Track` model attributes are optional. If there are no artist tags present, then, like any other attribute, `track.artists` is undefined. You need to handle this on your client code.

**Comment by jodal** [comment]
It seems that the code that triggered the bug was the example web app bundled with Mopidy.js, so I guess that could be made more robust.

**Comment by zoechi** [comment]
@kingosticks in https://github.com/mopidy/mopidy.js/blob/master/src/mopidy.d.ts#L227-L289 only `track_no`, `disc_no`, `length`, and `last_modified` are marked optional.

I just stumbled upon this because 
```js
{
  jsonrpc=2.0,
  id=0,
  method=core.tracklist.add,
  params={uris: [orfradio:oe3/live]},
}
```
returns
```js
{"jsonrpc": "2.0", "id": 0, "result": [{"__model__": "TlTrack", "tlid": 12, "track": {"__model__": "Track", "uri": "orfradio:oe3/live", "name": "Live"}}]}
```
where most fields are actually missing.
